core/src/main/kotlin/fr/sncf/osrd/graph/PathfindingGraph.kt

```diff
@@ -0,0 +1,47 @@
+package fr.sncf.osrd.graph
+
+import fr.sncf.osrd.sim_infra.api.Block
+import fr.sncf.osrd.stdcm.graph.extendLookaheadUntil
+import fr.sncf.osrd.stdcm.infra_exploration.InfraExplorer
+import java.util.Objects
+
+data class PathfindingEdge(
```

Comment:
Theorically, could put the `equals` and `hashcode` methods into the `InfraExplorer` and implement them for both `InfraExplorerImpl` and `InfraExplorerWithEnvelopeImpl`, completely trash this object and just create a `PathfindingGraph : Graph<InfraExplorer, InfraExplorer, Block>`.
Personnally, i think it's clearer likeso: states clearly that it's an edge, and we might not want such a loose equals and hashcode methods for `InfraExplorerImpl`. It's not a strong opinion though, feel free to give me your pov.

Reply:
I'm not a big fan of the semantics there, two `InfraExplorers` aren't equal just because they lead to the same place

Reply:
We agree on this then :)